ethpm/utils/registry.py

```diff
@@ -0,0 +1,48 @@
+from urllib import parse
+
+from typing import List
+
+from ethpm.exceptions import UriNotSupportedError
+
+from ethpm.validation import (
+    validate_package_name,
+    validate_registry_uri_authority,
+    validate_registry_uri_scheme,
+    validate_registry_uri_version,
+)
+
+
+def is_ens_domain(auth: str) -> bool:
+    """
+    Returns false if URI authority is not a valid ENS domain.
+    """
+    if auth[-4:] != '.eth' or len(auth.split('.')) not in [2, 3]:
+        return False
+    return True
+
+
+def is_registry_uri(uri: str) -> bool:
```

Comment:
If you split this into two functions then you'll be able to resolve the circular import issue your having.  This function either returns `True` or throws an exception which violates the principle of least surprise.  This should instead use all of the `is_` checks to return `True` or `False` and then write a `validate_registry_uri` function in your `validation.py` utils file that uses this.

Should allow you to remove all of the imports from this module that import from `validation` and to just have `validation` import from this module.


https://en.wikipedia.org/wiki/Principle_of_least_astonishment